src/fsharp/CompileOps.fs

```diff
@@ -2580,24 +2580,21 @@ type TcConfig private (data: TcConfigBuilder, validate: bool) =
             errorR(Error(FSComp.SR.buildMultipleReferencesNotAllowed libraryName, rangeCmdArgs)) 
             nameOfDll r
 
-    // Look for an explicit reference to mscorlib and use that to compute clrRoot and targetFrameworkVersion
+    // Look for an explicit reference to mscorlib/netstandard.dll or System.Runtime.dll and use that to compute clrRoot and targetFrameworkVersion
     let primaryAssemblyReference, primaryAssemblyExplicitFilenameOpt = computeKnownDllReference(data.primaryAssembly.Name)
-    let fslibReference, fslibExplicitFilenameOpt = 
-        let (_, fileNameOpt) as res = computeKnownDllReference getFSharpCoreLibraryName
+    let fslibReference =
+        // Look for explict FSharp.Core reference otherwise use version that was referenced by compiler
+        let dllReference, fileNameOpt = computeKnownDllReference getFSharpCoreLibraryName
```

Comment:
Where else is `computeKnownDllReference` used?  It seems odd that you're returning the first value if the second value was non-null (translated to C# because I think it illustrates the issue better):

``` c#
var someTuple = computeKnownDllReference(...);
if (someTuple.Item2 != null)
{
    // since you determined that ``.Item2`` is non-null, shouldn't that be what you return instead of ``.Item1``?
    return someTuple.Item1;
}
else
{
    // ...
}
```

Reply:
So this is what is happening the F# is really more concise:
````
    let fslibReference =
        // Look for explict FSharp.Core reference otherwise use version that was referenced by compiler
        let dllReference, fileNameOpt = computeKnownDllReference getFSharpCoreLibraryName
        match fileNameOpt with
        | Some _ -> dllReference
        | None -> AssemblyReference(range0, getDefaultFSharpCoreReference, None)
````
The code searches the referenced assemblies for FSharp.Core, if one was provided via -r: then fileNameOpt has a value, and we can just return the reference in dllReference.   Otherwise no fsharp.core was passed on the command line, so we use getDefaultFSharpCoreReference to find it and make an AssemblyReference and return that.


Previously we returned the assembly reference and the opt filename, but since we no longer use that, this refactoring just gets us the reference instead.